src/js_native_api.h

```diff
@@ -536,6 +536,8 @@ node_api_post_finalizer(node_api_basic_env env,
                         void* finalize_data,
                         void* finalize_hint);
 
+#warning "NAPI_EXPERIMENTAL is enabled. Experimental features may be unstable."
```

Comment:
This warning must be only in one place. It is better to be in the `js_native_api.h` file.
It should be closer to the top where we check/define the `NAPI_VERSION`.
We also must allow suppressing the warning by checking a special new macro.
The reason is that in some projects the warnings are treated as errors.

Reply:
Cool, thanks for that!
Does this macro need to be added in more places? Any pattern to follow?

Reply:
The macro must be just as a pre-condition for the warning to raise. It should be used only in a single place.
Its name can be something like: `NODE_API_EXPERIMENTAL_NO_WARNING`.
The important part here is the `NODE_API_` prefix.
Then the code can be as simple as:

```C++
#if defined(NAPI_EXPERIMENTAL) && !defined(NODE_API_EXPERIMENTAL_NO_WARNING)
#warning "NAPI_EXPERIMENTAL is enabled. Experimental features may be unstable."
#endif
```

By default users will see the warning when they use the `NAPI_EXPERIMENTAL`, but they can suppress it if they add the `NODE_API_EXPERIMENTAL_NO_WARNING` definition to the compiled code.

Reply:
I added like you suggested, thanks!